### PR TITLE
Code reworked to make insert and fetch timings more stable

### PR DESCRIPTION
**Pull request by T0ha**

### CODE REVIEW

File: lib/telemetry_metrics_mnesia/db.ex [hunk 1]
```diff
@@ -24,17 +26,39 @@ defmodule TelemetryMetricsMnesia.Db do
...
-      )
+      for metric <- metrics,
+          metric.event_name == event,
+          is_nil(metric.keep) or metric.keep.(metadata) do
+        data =
+          {@telemetry_metrics_table, metric}
+          |> Mnesia.wread()
+          |> case do
+            [] when metric.tags == [] -> []
+            [] -> []
```
Comment (v0idpwn) [inline lib/telemetry_metrics_mnesia/db.ex]:
```suggestion
            [] -> []
```

File: lib/telemetry_metrics_mnesia/db.ex [hunk 2]
```diff
@@ -43,36 +67,112 @@ defmodule TelemetryMetricsMnesia.Db do
...
+
+          v, acc ->
+            reducer.(v, acc)
+        end
+    end
+  end
+
+  defp granularity_min_timestamp(%_{reporter_options: opts}) do
+    case Keyword.get(opts, :granularity) do
+      [{unit, amount}] ->
```
Comment (v0idpwn) [inline lib/telemetry_metrics_mnesia/db.ex:126]: Why is granularity a single element list? Should it be `{unit, amount}` instead? Is it possible to have >1 element?

File: lib/telemetry_metrics_mnesia/event_handler.ex [hunk 3]
```diff
@@ -34,11 +37,27 @@ defmodule TelemetryMetricsMnesia.EventHandler do
...
+  end
+
+  def start_link(_), do: GenServer.start_link(__MODULE__, [], name: __MODULE__)
+
+  @impl true
+  def init(_), do: {:ok, true}
+
+  @impl true
+  def handle_call({:event, event, measurements, metadata, metrics}, _, _) do
+    Db.write_event(event, measurements, metadata, metrics)
```
Comment (v0idpwn) [inline lib/telemetry_metrics_mnesia/event_handler.ex:42]: Why the move to run `Db.write_event/4` in the server and not in the telemetry handler?

File: lib/telemetry_metrics_mnesia/db.ex (hunk 2 above)
Comment (T0ha) [reply lib/telemetry_metrics_mnesia/db.ex:126]: This is don for more beytiful `Keyword`-like syntax, eg. `[granularuty: [milliseconds: 10]]`. But tuples are fine, but seems less conventional.

Comment (v0idpwn) [reply lib/telemetry_metrics_mnesia/db.ex:126]: ah, I didn't realize, makes sense!

**Pull request merged**